sql/core.go

```diff
@@ -370,14 +371,56 @@ type SystemVariableRegistry interface {
 	NewSessionMap() map[string]SystemVarValue
 	// GetGlobal returns the global value of the system variable with the given name
 	GetGlobal(name string) (SystemVariable, interface{}, bool)
+	// GetResetVal returns the default value of the system variable with the given name.
+	// This is not the current GLOBAL value. It's used for postgres config parameters only.
+	GetResetVal(name string) (SystemVariable, interface{}, bool)
 	// SetGlobal sets the global value of the system variable with the given name
 	SetGlobal(name string, val interface{}) error
 	// GetAllGlobalVariables returns a copy of all global variable values.
 	GetAllGlobalVariables() map[string]interface{}
 }
 
-// SystemVariable represents a system variable.
-type SystemVariable struct {
+// SystemVariable is used to handle mysql system variables and postgres configuration variables interchangeably.
+// TODO: mysql sv and postgres cp should not be mixed.
+type SystemVariable interface {
+	// GetName returns the name of the sv. Case-sensitive.
+	GetName() string
+	// SetName sets the name of the sv.
+	// TODO: it should not happen, but we set the name to lower-case string?
+	SetName(string)
+	// GetType returns the type of the sv.
+	GetType() Type
+	// SetDefault sets the default value of the sv.
+	SetDefault(any)
+	// GetDefault returns the default value of the sv.
+	GetDefault() any
+	// GetValue returns the value
+	GetValue(a any) (any, bool)
+	// HasDefaultValue checks whether the default value
+	// of the sv is the same value of given argument.
+	HasDefaultValue(any) bool
```

Comment:
It would be better to only have `GetDefault()`, and let the user compare it with their value manually.